diff --git a/src/lib/Hydra/Plugin/GitlabPulls.pm b/src/lib/Hydra/Plugin/GitlabPulls.pm
index b44a1a98..24811454 100644
--- a/src/lib/Hydra/Plugin/GitlabPulls.pm
+++ b/src/lib/Hydra/Plugin/GitlabPulls.pm
@@ -75,7 +75,7 @@ sub fetchInput {
     # Get the target project URL, as it is the one we need to build the pull
     # urls from later
     (my $repo, my $res) = _query("$baseUrl/api/v4/projects/$projectId", $ua);
-    my $target_repo_url = $repo->{http_url_to_repo};
+    my $target_repo_url = $repo->{ssh_url_to_repo};
 
     _iterate($url, $baseUrl, \%pulls, $ua, $target_repo_url);
 

